osi_featuredata.proto

```diff
@@ -557,6 +563,398 @@ message UltrasonicIndirectDetection
+    // SensorDetectionHeader::measurement_time.
+    //
+    // The timestamp of this detection :=
+    // \c SensorDetectionHeader::measurement_time + \c #time_difference.
+    //
+    optional Timestamp time_difference = 3;
+
+    // Definition of the image shape type of this detection.
+    //
+    optional ImageShapeType image_shape_type = 4;
+
+    // The defined shape is background.
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    // Range: [0.0, 1.0]
+    //
+    optional bool shape_classification_background = 5;
+
+    // The defined shape is foregroud.
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_foreground = 6;
+
+    // The defined shape is flat.
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_flat = 7;
+
+    // The defined shape is upright.
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_upright = 8;
+
+    // The defined shape is ground.
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_ground = 9;
+
+    // The defined shape is sky.
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_sky = 10;
+
+    // The defined shape is vegetation.
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_vegetation = 11;
+
+    // The defined shape is a road.
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_road = 12;
+
+    // The defined shape is a non-driving lane (e.g. sidewalk).
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_non_driving_lane = 13;
+
+    // The defined shape is non-road (e.g. traffic island).
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_non_road = 14;
+
+    // The defined shape is a stationary object.
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_stationary_object = 15;
+
+    // The defined shape is a possible moving object.
+    // The probability for this classification is at least
+    // \c #shape_classification_probability.
+    //
+    optional bool shape_classification_possible_moving_object = 16;
```

Comment:
I woukd remove the "possible" here because of the probabilities everything is "possible"?